compatibility/sandbox-migration/runner/Migration/Types.hs

```diff
@@ -29,6 +30,17 @@ data Test s r = Test
   -- ^ The initial state to start with.
   }
 
+interleave :: Test s r -> Test s' r' -> Test (s, s') (r, r')
```

Comment:
I'm about to add a third test, would it be better to interleave a list of tests.

Reply:
You can just nest this, `x `interleave` y `interleave` z` or if you have a list use a fold for it.